consensus/application.go

```diff
@@ -432,6 +432,7 @@ func (ms *MidState) reviseFileContractElement(fce types.FileContractElement, rev
 	} else if fced.Revision != nil {
 		*fced.Revision = rev
 	} else {
+		dupProof(&fce.StateElement)
```

Comment:
never could I possibly have predicted that Merkle proof aliasing would bite me (again (again))

Reply:
Should definitely just go ahead and `dupProof` anywhere with a Merkle tree regardless of if it's currently an issue lol.